pkg/meta/repodb/common/common.go

```diff
@@ -62,43 +62,89 @@ func ValidateRepoReferenceInput(repo, reference string, manifestDigest godigest.
 	return nil
 }
 
-func ScoreRepoName(searchText string, repoName string) int {
+// These constants are meant used to describe how high or low in rank a match is.
+// Note that the "higher rank" relates to a lower number so ranks are sorted in a
+// ascending order.
+const (
+	lowPriority          = 100
+	mediumPriority       = 10
+	highPriority         = 1
+	perfectMatchPriority = 0
+)
+
+// RankRepoName associates a rank to a given repoName given a searchText.
+// The imporance of the value grows inversly proportional to the int value it has.
+// For example: rank(1) > rank(10) > rank(100)...
+func RankRepoName(searchText string, repoName string) int {
+	searchText = strings.Trim(searchText, "/")
 	searchTextSlice := strings.Split(searchText, "/")
 	repoNameSlice := strings.Split(repoName, "/")
 
+	// ex: zot/test/alpine, "alpine"
```

Comment:
"repo/repo1/abc" use other names
or remove the comment, if there are examples in test cases